agent/task_context.go

```diff
@@ -691,18 +701,62 @@ func (tc *taskContext) getAddMetadataTagResponse() *triggerAddMetadataTagResp {
 	return tc.addMetadataTagResp
 }
 
-func (tc *taskContext) getDeviceNames(ctx context.Context) error {
-	if tc.taskConfig == nil || tc.taskConfig.Distro == nil || len(tc.taskConfig.Distro.Mountpoints) == 0 {
-		return nil
+// getPlatformMountpoint returns the standard mountpoint used for the current OS.
+func getPlatformMountpoint() string {
+	switch runtime.GOOS {
+	case "linux":
+		return linuxMountpoint
+	case "windows":
+		return windowsMountpoint
+	case "darwin":
+		return darwinMountpoint
+	default:
+		return ""
+	}
+}
+
+// getMountpoints returns the list of mountpoints to check for disk devices.
+// Always includes "/" as the first mountpoint, followed by platform-specific mountpoint if available.
+func (tc *taskContext) getMountpoints() []string {
+	mountpoints := []string{"/"}
+	if platformMountpoint := getPlatformMountpoint(); platformMountpoint != "" {
+		mountpoints = append(mountpoints, platformMountpoint)
 	}
+	return mountpoints
+}
+
+// getDeviceName extracts and sanitizes a device name from a device path.
+func getDeviceName(device string) string {
+	deviceName := filepath.Base(device)
+
+	// filepath.Base of windows drive letters could return these
+	if deviceName == "" || deviceName == "/" || deviceName == "\\" {
+		// check if partition device is a windows drive
+		if runtime.GOOS == "windows" && len(device) == 2 && device[1] == ':' && device[0] >= 'A' && device[0] <= 'Z' {
```

Comment:
This line looks a bit confusing, could we break it off to a helper to be more clear?

Reply:
Good point, created a helper function